### ISSUE TITLE
Nikon Z 9

### ISSUE THREAD

**Issue by peti** [darktable-org/darktable#10889]
I have uploaded three sample NEF files created by the Nikon Z 9 camera to https://raw.pixls.us/:

- `nikon-z9-raw-lossless.nef`
- `nikon-z9-raw-compressed-high-quality.nef`
- `nikon-z9-raw-compressed-low-quality.nef`

The Z 9 does not support uncompressed RAW. The camera knows only "RAW L" with different levels of compression. The compression modes are:

- "loss-less",
- "high efficiency*", and 
- "high efficiency",

where the star variant ("`*`") denotes higher quality a.k.a. less efficient compression.

The images show an X-Rite Color Checker Passport.

I have not touched the files except renaming them for clarity; these come straight out of the camera.

**Comment by kmilos** [comment]
https://onlinemanual.nikonimglib.com/z9/en/15_menu_guide_02_09.html#nef_raw_recording

**Comment by fxgsell** [comment]
hi everyone, any timeline on that? Currently limiting myself to jpeg to use dt.

**Comment by rr-** [comment]
For the time being, one workable solution is to use Rawtherapee – it can at least open the new .NEFs, even if the initial curves are completely off and require tons of adjusting.

**Comment by kmilos** [comment]
Just use the dt development branch, and change cameras.xml as shown here, it's not that complicated really.

**Comment by rr-** [comment]
Thanks, but I don't want to spend an afternoon griping with DT's build system which has been [failing for the entire past week](https://github.com/darktable-org/darktable/actions/workflows/ci.yml?query=branch%3Amaster+is%3Acompleted+event%3Apush) especially that it means a migration of all of the sidecar files with no hope of returning (unless I spend even more time on getting stable + develop builds on my computer in parallel).

**Comment by fxgsell** [comment]
Thanks, that works!

**Comment by ValZapod** [comment]
Here are N-RAW video from 2.0 firmaware! Cool! https://forum.grassvalley.com/forum/editors/editing-with-edius/572094-n-raw-from-z9-sample-download

**Comment by robcampbell95** [comment]
I tried adding that patch but it didn't work.  I am using the FTZ II adapter if that makes a difference.  The lens doesn't show properly.  The lens reports: VR 70-300mm f/4.5-5.6E but it should be: Nikon AF-P Nikkor 70-300mm f/4.5-5.6E ED VR.

Do I need to submit samples?

**Comment by gitmatt19** [comment]
> I tried adding that patch but it didn't work. I am using the FTZ II adapter if that makes a difference. The lens doesn't show properly. The lens reports: VR 70-300mm f/4.5-5.6E but it should be: Nikon AF-P Nikkor 70-300mm f/4.5-5.6E ED VR.
> 
> Do I need to submit samples?

Looks there the text you have is in the EXIF data but there are multiple lens strings that could be used....

Nikon 200-500 with FTZ2 :

$ exiftool '/DSC_7126.JPG'  | grep -i lens
Lens Type                       : E VR
Lens                            : 200-500mm f/5.6
Lens F Stops                    : 5.00
Lens Data Version               : 0802
Lens Info                       : 200-500mm f/5.6
Lens Make                       : 
Lens Model                      : VR 200-500mm f/5.6E
Lens Spec                       : 200-500mm f/5.6 E VR
Lens ID    

Nikon Z 100-400 :
$ exiftool 'DSC_7118.JPG'  | grep -i lens
Lens Type                       : VR
Lens                            : 100-400mm f/4.5-5.6
Lens F Stops                    : 5.67
Lens Data Version               : 0802
Lens Info                       : 100-400mm f/4.5-5.6
Lens Make                       : NIKON
Lens Model                      : NIKKOR Z 100-400mm f/4.5-5.6 VR S
Lens Spec                       : 100-400mm f/4.5-5.6 VR
Lens ID                         : NIKKOR Z 100-400mm f/4.5-5.6 VR S

**Comment by kmilos** [comment]
Lens detection is an entirely separate issue, please do not add "noise" here. See https://github.com/darktable-org/darktable/issues/6317 and https://github.com/darktable-org/darktable/issues/10053

**Comment by kmilos** [comment]
Support has now been added to exiv2 to read the lens data for the Z 9, in both main and 0.27-maintenance branches.
